point_viewer_grpc/src/service.rs

```diff
@@ -36,7 +37,7 @@ use std::str::FromStr;
 use std::sync::{Arc, RwLock};
 
 struct OctreeServiceData {
-    octree: Box<Octree>,
+    octree: [Octree; 1], // caveat: only one octree in this slice
```

Comment:
You could make BatchIterator take a `&[&Octree]` instead of an `&[Octree]`, which would allow you to avoid these changes. Then below, you'd write
```
                let octree_slice = &[&service_data.octree];
                let mut batch_iterator = BatchIterator::new(
                    octree_slice,
                    &query,
                    num_points_per_batch,
                    num_cpus::get() - 1,
                );
```

Reply:
`&[octree]` seems to be more natural to use to me. Here is an exception, but e.g. in the point cloud client we have a vector of octrees and can just reference it...

Reply:
That's fine by me too.

Reply:
than if I understood well, i would let this code as is

Reply:
Yes, thanks.